rust/projects/tools/answers.md

```diff
@@ -5,19 +5,86 @@
 > What does this directory layout suggest about how `lib.rs`, `main.rs`, and
 `tests.rs` are compiled, and how they are linked to each other?
 
-Answer: TODO
+Answer: The compilation of our library file `lib.rs` is independent from
+`main.rs` and `tests.rs`. The library becomes an implicit dependency of
+`main.rs` and `tests.rs`, and is linked to the binaries `main.rs` and
+`tests.rs` compile into.
```

Comment:
It might be worth mentioning that cargo decides what to do with all three of these files implicitly, that lib.rs is compiled into a library because cargo recognizes `src/lib.rs` specially; that `main.rs` is compiled into an executable because cargo interprets any rust file in `src/bin/` as an executable; `tests.rs` is compiled into test binary via `cargo test` because cargo treats any source file in ... etc.